src/Integration/Binding/ProjectBinderFactory.cs

```diff
@@ -18,19 +18,37 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System;
+using System.IO.Abstractions;
 using System.Linq;
 using EnvDTE;
 
 namespace SonarLint.VisualStudio.Integration.Binding
 {
     internal class ProjectBinderFactory : IProjectBinderFactory
     {
+        private readonly IServiceProvider serviceProvider;
+        private readonly IFileSystem fileSystem;
+
+        public ProjectBinderFactory(IServiceProvider serviceProvider)
+            : this(serviceProvider, new FileSystem())
+        {
+        }
+
+        internal ProjectBinderFactory(IServiceProvider serviceProvider, IFileSystem fileSystem)
+        {
+            this.serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
         public IProjectBinder Get(Project project)
         {
             var languages = ProjectToLanguageMapper.GetAllBindingLanguagesForProject(project).ToList();
             var isRoslynProject = languages.Contains(Core.Language.VBNET) || languages.Contains(Core.Language.CSharp);
 
-            return isRoslynProject ? new RoslynProjectBinder() : null;
+            return isRoslynProject
+                ? (IProjectBinder) new RoslynProjectBinder(serviceProvider, fileSystem)
```

Comment:
Is the cast required?

Reply:
Yep, otherwise it will throw "conditional expression cannot be determined because there is no implicit conversion"